canary/MeasureTransferRate.cpp

```diff
@@ -0,0 +1,484 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "MeasureTransferRate.h"
+#include "CanaryApp.h"
+#include "CanaryUtil.h"
+#include "MetricsPublisher.h"
+#include "S3ObjectTransport.h"
+#include <aws/common/clock.h>
+#include <aws/common/system_info.h>
+#include <aws/crt/http/HttpConnection.h>
+#include <aws/crt/http/HttpConnectionManager.h>
+#include <aws/crt/http/HttpRequestResponse.h>
+#include <condition_variable>
+#include <mutex>
+
+using namespace Aws::Crt;
+
+namespace
+{
+    const uint64_t SinglePartObjectSize = 5ULL * 1024ULL * 1024ULL * 1024ULL;
```

Comment:
These all feel like config options with these values as the defaults.

Reply:
Agreed!  Once we have a proper config file that we load I'll add this and the others there.  Right now everything is CLI which is already getting a bit crowded for options.